urlaf656038a0_Salihan04_MOBLIMA_tgz-pJ8-com_moblima_util_dataTestJ8/src/com/moblima/presentation/ManageShowTimes_CLI.java

```diff
@@ -0,0 +1,275 @@
+package com.moblima.presentation;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Enumeration;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Scanner;
+
+import com.moblima.businesslogic.CinemaBL;
+import com.moblima.businesslogic.CineplexBL;
+import com.moblima.businesslogic.MovieBL;
+import com.moblima.businesslogic.ShowTimeBL;
+import com.moblima.dataaccess.CinemaDAOImpl;
+import com.moblima.dataaccess.CineplexDAOImpl;
+import com.moblima.dataaccess.MovieDAO;
+import com.moblima.dataaccess.MovieDAOImpl;
+import com.moblima.model.Cinema;
+import com.moblima.model.Cineplex;
+import com.moblima.model.Movie;
+import com.moblima.model.ShowTime;
+import com.moblima.util.ConsoleReader;
+
+public class ManageShowTimes_CLI {
+	private CineplexBL cineplexBL;
+	private CinemaBL cinemaBL;
+	private MovieBL movieBL;
+	private ShowTimeBL showTimeBL;
+
+	private List<ShowTime> _showTimes;
+
+	public ManageShowTimes_CLI() {
+		cineplexBL = new CineplexBL();
+		movieBL = new MovieBL();
+		cinemaBL = new CinemaBL();
+		showTimeBL = new ShowTimeBL();
+	}
+
+	public void listShowTimes() {
+		int index = 0;
+		_showTimes = new ArrayList<ShowTime>(); // To be used for update and remove showTime
+
+		System.out.println("ShowTimes");
+		System.out.println("=========");
+
+		for(Cineplex cineplex: cineplexBL.getCineplexes()) {
+			List<Movie> movies = cineplex.getMovies();
+
+			if(movies.size() == 0) {
+				continue;
+			}
+
+			System.out.println(cineplex.getCineplexName());
+			System.out.println("-----------------------------------");
+
+			for(Movie movie: movies) {				
+				List<ShowTime> showTimes = showTimeBL.getShowTimes(movie, cineplex);
+
+				if(showTimes.size() == 0) {
+					continue;
+				}
+
+				System.out.println(movie.getTitle());
+				System.out.println("********************************");
+
+				Hashtable<String, List<ShowTime>> showTimeHashTable = new Hashtable<String, List<ShowTime>>();
+				//Preprocessing for showTime to follow the cathay format
+				for(ShowTime st: showTimes) {
+					String d = new SimpleDateFormat("d MMM, E").format(st.getTime());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.